feat(tactic/rename_var): A teaching tactic (#1974)
leanprover-community/mathlib: docs/tactics.md

This can be used to simulate the `specialize` and `apply at` tactics
of Coq.

## rename_var

> `rename_var old new` renames variable `old` to `new` in the goal.

Reviewer: ```suggestion
`rename_var old new` renames all bound variables named `old` to `new` in the goal.
```

`rename_var old new at h` does the same in hypothesis `h`.

> This is meant for teaching bound variables only, Lean will never need such a renaming.

Reviewer: ```suggestion
This is meant for teaching bound variables only. Such a renaming should never be relevant to Lean.
```
?
